src/main/scala/org/kframework/minikore/Constructors.scala

```diff
@@ -0,0 +1,97 @@
+package org.kframework.minikore
+
+import org.kframework.minikore.MiniKore._
+
+
+// Constructor Factory.
```

Comment:
I personally do not like this idea to group all the constructors into one place, separate from the destructors.  Since abstract classes allow constructors, I would rather have an abstract class corresponding to all the (mini)kore language/data-structures and then traits for other views of them.  In other words, instead of having traits for all interfaces, including the basic kore syntax/language/constructs as well as the Ast view, I would use an abstract class for the "main trait", with constructors in it which must be therefore enforced to be implemented, and traits for anything else.  This should also facilitate interfacing with Java (which seems to have problems with traits providing some default implementations).  Or is it exactly what you are doing and I'm missing something?